traffic_ops/testing/api_contract/v4/conftest.py

```diff
@@ -336,3 +406,68 @@ def cdn_post_data(to_session: TOSession, cdn_prereq_data: list[JSONData]) -> dic
 	except IndexError:
 		logger.error("No CDN response data from cdns POST request.")
 		sys.exit(1)
+
+
+@pytest.fixture()
+def cachegroup_post_data(to_session: TOSession, request_template_data: list[JSONData]
+			 ) -> dict[str, object]:
+	"""
+	PyTest Fixture to create POST data for cachegroup endpoint.
+
+	:param to_session: Fixture to get Traffic Ops session.
+	:param request_template_data: Fixture to get Cachegroup data from a prerequisites file.
+	:returns: Sample POST data and the actual API response.
+	"""
+	try:
+		cachegroup = request_template_data[0]
+	except IndexError as e:
+		raise TypeError(
+			"malformed prerequisite data; no Cache group present in 'cachegroup' array property") from e
+
+	if not isinstance(cachegroup, dict):
+		raise TypeError(
+			f"malformed prerequisite data; Cache group must be objects, not '{type(cachegroup)}'")
+
+	# Return new post data and post response from cachegroups POST request
+	randstr = str(randint(0, 1000))
+	try:
+		name = cachegroup["name"]
+		if not isinstance(name, str):
+			raise TypeError(f"name must be str, not '{type(name)}'")
+		cachegroup["name"] = name[:4] + randstr
+		short_name = cachegroup["shortName"]
+		if not isinstance(short_name, str):
+			raise TypeError(f"shortName must be str, not '{type(short_name)}")
+		cachegroup["shortName"] = short_name[:5] + randstr
+	except KeyError as e:
+		raise TypeError(f"missing Cache group property '{e.args[0]}'") from e
+	# Hitting types GET method to access typeID for cachegroup POST data
+	type_get_response: tuple[
+		dict[str, object] | list[dict[str, object] | list[object] | primitive] | primitive,
+		requests.Response
+	] = to_session.get_types(query_params={"useInTable": "cachegroup"})
+	try:
+		type_data = type_get_response[0]
+		if not isinstance(type_data, list):
+			raise TypeError("malformed API response; 'response' property not an array")
+		first_type = type_data[0]
+		if not isinstance(first_type, dict):
+			raise TypeError("malformed API response; first Type in response is not an object")
+		cachegroup["typeId"] = first_type["id"]
+		type_id = cachegroup["typeId"]
+		logger.info("extracted %s from %s", type_id, type_get_response)
+	except KeyError as e:
+		raise TypeError(f"missing Type property '{e.args[0]}'") from e
+
+	logger.info("New cachegroup data to hit POST method %s", request_template_data)
+	# Hitting cachegroup POST method
+	response: tuple[JSONData, requests.Response] = to_session.create_cachegroups(data=cachegroup)
+	try:
+		resp_obj = response[0]
+		if not isinstance(resp_obj, dict):
+			raise TypeError("malformed API response; cache group is not an object")
+		return resp_obj
+	except IndexError:
+		logger.error("No Cache group response data from cdns POST request.")
+		sys.exit(1)
+
```

Comment:
There's an extra trailing newline here, and it's the only reason Pylint gives your code less than a perfect score.

Reply:
Removed trailing newline.